client/components/tinymce/plugins/wpcom-view/wpvideo-view-container.jsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import MediaActions from 'lib/media/actions';
+import MediaListStore from 'lib/media/list-store';
+import WpVideoView from './wpvideo-view';
+
+const getStateData = siteId => {
+	return {
+		media: MediaListStore.getAll( siteId ),
+		mediaHasNextPage: MediaListStore.hasNextPage( siteId ),
+		mediaFetchingNextPage: MediaListStore.isFetchingNextPage( siteId )
+	};
+};
+
+class WpVideoViewContainer extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = getStateData( props.siteId );
+	}
+
+	componentDidMount() {
+		MediaListStore.on( 'change', this.updateStateData.bind( this ) );
+		this.fetchData( this.props.siteId );
+	}
+
+	componentWillUnmount() {
+		MediaListStore.off( 'change', this.updateStateData.bind( this ) );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.fetchData( this.props.siteId );
+			this.updateStateData();
+		}
+	}
+
+	fetchData( siteId ) {
+		setTimeout( () => {
```

Comment:
It may also be worth exploring whether we should consider adding support for fetching individual video files from the REST API, such that we don't need to manage querying nor request the entire set of media.

See: https://developer.wordpress.com/docs/api/1.1/get/videos/%24guid/